CDAP-6100 Support special characters in documentation "tabbed-parsed-literal" tab labels.

awholegunch opened the pull request:
Passes [Quick Build 4](http://builds.cask.co/browse/CDAP-DQB30-4)

[CLI Commands](http://builds.cask.co/artifact/CDAP-DQB30/shared/build-4/Docs-HTML/3.5.0-SNAPSHOT/en/reference-manual/cli-api.html)

Fix for https://issues.cask.co/browse/CDAP-6100

This does not demo using special characters, but instead shows that nothing breaks...the CLI commands show an example of tabs that include spaces and use a mapping that interacts with the standard "linux-windows" mapping.

Commit:
Support special characters in tab labels.

Commit:
Missed an instance that requires being lower.

Commit:
Merge branch 'develop' into docs/3.5_Doc_changes

# Conflicts:
#	cdap-docs/_common/tabbed-parsed-literal.py
# resolved.

bdmogal commented on cdap-docs/_common/tabbed-parsed-literal.py:
```diff
@@ -226,6 +228,21 @@ def dequote(s):
+        s_clean += c if c.isalnum() else '-'
+    return s_clean
+
+def cleaned_string(s):
```
method names should usually be verbs?

awholegunch replied:
Removed, as unused.

bdmogal commented on cdap-docs/_common/tabbed-parsed-literal.py:
```diff
@@ -226,6 +228,21 @@ def dequote(s):
+    If a string has any non-ASCII characters, replace them with a hyphen.
+    """
+    s_clean = ''
+    for c in s:
```
better variable names, perhaps? Although I can sort of guess that this is `for char in string`?

awholegunch replied:
Fixed.

bdmogal commented on cdap-docs/_common/tabbed-parsed-literal.py:
```diff
@@ -376,23 +393,30 @@ def cleanup_content(self):
             return default
         else:
-            return dequote(_option)
+            return clean_ascii(dequote(_option)) if ascii_only else dequote(_option)
```
can just use `cleaned_string(_option) if ascii_only else dequote(_option)`

awholegunch replied:
Umm. No. We want to first de-quote the option, and then replace any remaining non-alphanumeric characters with hyphens. If you do as you suggest, you'll end up with this:

```
-I-was-enclosed-in-quotes-
```

instead of 

```
I-was-enclosed-in-quotes
```

bdmogal replied:
ok

bdmogal commented on cdap-docs/_common/tabbed-parsed-literal.py:
```diff
@@ -630,7 +658,7 @@ def warner(msg):
                                                     
     nav_tabs_html = NAV_TABS % nav_tabs_html
-    tab_content_html = TAB_CONTENT_START + tab_content_html + DIV_END
+    tab_content_html = TAB_CONTENT_START + tab_content_html + DIV_END    
```
nit: no need for space/tab in the end?

awholegunch replied:
Fixed.

bdmogal commented:
@awholegunch left some minor comments. This PR itself LGTM. But I observed while reviewing the cli page http://builds.cask.co/artifact/CDAP-DQB30/shared/build-3/Docs-HTML/3.5.0-SNAPSHOT/en/reference-manual/cli-api.html that the commands for distributed CDAP and Cloudera Manager are wrong. They start with a period (`./opt/cdap...` or ./opt/cloudera``) which indicates that they are relative paths when they should be absolute. The period should be removed. Not sure if that would be a different PR though.

Commit:
Changed variable names to reflect what they are better. Removed an un-used
wrapper function ("cleaned_string"). Replaced usage of "ASCII" with
"alphanumeric" as a more accurate description.

Commit:
Fix Distributed and CM CLI command examples.

awholegunch commented:
Fixed the CLI examples for Distributed and Cloudera Manager: http://builds.cask.co/artifact/CDAP-DQB30/shared/build-4/Docs-HTML/3.5.0-SNAPSHOT/en/reference-manual/cli-api.html#interactive-mode

bdmogal commented:
LGTM
